### PR TITLE
Framework for fetching CDO data 

### PR DESCRIPTION
**Pull request by mjoyce91**
- Ability to select a fellow CDO and fetch their clients
- Uses mock responses in actual requests, instead of everything being hard-coded. Will still need to update these to real endpoints. So, the requests will fail as expected when you select a new CDO.

### CODE REVIEW

File: src/Components/BidderPortfolio/CDOAutoSuggest/__snapshots__/CDOAutoSuggest.test.jsx.snap [hunk 1]
```diff
@@ -23,7 +23,7 @@ exports[`CDOAutoSuggest matches snapshot 1`] = `
       onClick={[Function]}
       type="div"
     >
-      Leah Shadtrach
+      undefined undefined
```
Comment (mjoyce91) [inline src/Components/BidderPortfolio/CDOAutoSuggest/__snapshots__/CDOAutoSuggest.test.jsx.snap]: I'm going to update this snapshot test

**Comment by mjoyce91** [comment]
Updates rely on https://github.com/MetaPhase-Consulting/State-TalentMAP-API/pull/250 and https://github.com/MetaPhase-Consulting/State-TalentMAP-API/pull/252

File: src/actions/bidList.js [hunk 2]
```diff
@@ -176,7 +176,7 @@ export function clientBidListFetchData(ordering = 'draft_date') {
   return (dispatch, getState) => {
     if (shouldUseClient(getState)) {
       const { id } = getState().clientView.client;
```
Comment (rtirserio) [inline src/actions/bidList.js]: @mjoyce91 the `id` value in the URL needs to be the users emp_id value

Comment (mjoyce91) [reply src/actions/bidList.js]: @rtirserio looks like the API is currently configured to take the perdet

**Comment by rtirserio** [comment]
yes, in our database thats the emp_id. I wasnt sure what value you were
pulling the id from

**Comment by mjoyce91** [comment]
Test locally by setting the feature flag `available_positions: true`

**Comment by mjoyce91** [comment]
@rtirserio this is good to test for complete integration. Currently I'm doing a JS `find` by perdet in lieu of a `/client/:id` endpoint, so we can add that later.

**Comment by rtirserio** [comment]
@mjoyce91 ive added the `client/:id` endpoint in https://github.com/MetaPhase-Consulting/State-TalentMAP-API/pull/257

File: src/Containers/ProfilePublic/ProfilePublic.jsx [hunk 3]
```diff
@@ -21,7 +21,7 @@ class ProfilePublic extends Component {
     const { assignments, bidList } = userProfile;
     const assignment = get(assignments, '[0]');
     return (
-      hasErrored ?
+      (hasErrored && 1 === false) ?
```
Comment (mjoyce91) [inline src/Containers/ProfilePublic/ProfilePublic.jsx]: Need to fix this

**Comment by rtirserio** [comment]
@mjoyce91 i updated the API to return the `perdet_seq_num` as the id value since that was causing issues with react and keys. (https://github.com/MetaPhase-Consulting/State-TalentMAP-API/pull/262)

I am not seeing the filtering working. when i select a CDO i get all the clients

**Comment by mjoyce91** [comment]
@rtirserio - updated filtering by CDO (relies on https://github.com/MetaPhase-Consulting/State-TalentMAP-API/pull/263)

**Comment by rtirserio** [comment]
@mjoyce91 the filtering looks good.

See the screenshot below, probably only an issue with a bunch of skills...

Screen Shot 2019-12-11 at 1 26 50 PM

**Comment by mjoyce91** [comment]
@Rob - that's fixed in #595

**Comment by rtirserio** [comment]
@mjoyce91 ok all looks good here just the failing test

**Comment by mjoyce91** [comment]
@rtirserio updated

**Pull request merged**